tests/vmi_multus_test.go

```diff
@@ -876,29 +876,162 @@ var _ = Describe("[Serial]SRIOV", func() {
+		vmi := newCirrosVMIWithMacvtapNetwork(networkName)
+		vmi.Spec.Domain.Devices.Interfaces[0].MacAddress = mac
+
+		return tests.RunVMIAndExpectLaunchWithIgnoreWarningArg(vmi, 180, false)
+	}
+
+	createCirrosVMIWithMacvtapStaticIP := func(virtClient kubecli.KubevirtClient, nodeName string, networkName string, ifaceName string, ipCIDR string, mac *string) *v1.VirtualMachineInstance {
+		vmi := newCirrosVMIWithMacvtapNetwork(networkName)
+		if mac != nil {
+			vmi.Spec.Domain.Devices.Interfaces[0].MacAddress = *mac
+		}
+		vmi = tests.WaitUntilVMIReady(
+			tests.StartVmOnNode(vmi, nodeName),
+			tests.LoggedInCirrosExpecter)
+		// configure the client VMI
+		configVMIInterfaceWithSudo(vmi, ifaceName, ipCIDR)
+		return vmi
+	}
+
+	Context("a virtual machine with one macvtap interface, with a custom MAC address", func() {
+		var serverVMI *v1.VirtualMachineInstance
+		var chosenMAC string
+		var serverCIDR string
+		var nodeList *k8sv1.NodeList
+		var nodeName string
+
+		BeforeEach(func() {
+			nodeList = tests.GetAllSchedulableNodes(virtClient)
+			Expect(nodeList.Items).NotTo(BeEmpty(), "schedulable kubernetes nodes must be present")
+			nodeName = nodeList.Items[0].Name
+			chosenMAC = "de:ad:00:00:be:af"
+			serverCIDR = "192.0.2.102/24"
+
+			serverVMI = createCirrosVMIWithMacvtapStaticIP(virtClient, nodeName, macvtapNetworkName, "eth0", serverCIDR, &chosenMAC)
+		})
+
+		It("should have the specified MAC address reported back via the API", func() {
+			Expect(len(serverVMI.Status.Interfaces)).To(Equal(1), "should have a single interface")
+			Expect(serverVMI.Status.Interfaces[0].MAC).To(Equal(chosenMAC), "the expected MAC address should be set in the VMI")
+		})
+
+		Context("and another virtual machine connected to the same network", func() {
+			var clientVMI *v1.VirtualMachineInstance
+			BeforeEach(func() {
+				clientVMI = createCirrosVMIWithMacvtapStaticIP(virtClient, nodeName, macvtapNetworkName, "eth0", "192.0.2.101/24", nil)
+			})
+
+			It("can communicate with the virtual machine in the same network", func() {
+				Expect(libnet.PingFromVMConsole(clientVMI, cidrToIP(serverCIDR))).To(Succeed())
+			})
+		})
+	})
+
+	Context("VMI migration", func() {
+		var macvtapVMI *v1.VirtualMachineInstance
+
+		BeforeEach(func() {
+			nodes := tests.GetAllSchedulableNodes(virtClient)
+			Expect(nodes.Items).ToNot(BeEmpty(), "There should be some compute node")
+
+			if len(nodes.Items) < 2 {
+				Skip("Migration tests require at least 2 nodes")
+			}
+
+			if !tests.HasLiveMigration() {
+				Skip("Migration tests require the 'LiveMigration' feature gate")
+			}
+		})
+
+		BeforeEach(func() {
+			macAddress := "02:03:04:05:06:07"
+			macvtapVMI = createCirrosVMIWithMacvtapDefinedMAC(virtClient, macvtapNetworkName, macAddress)
+		})
+
+		It("should be successful when the VMI MAC address is defined in its spec", func() {
```

Comment:
It is a bit confusing. If the mac is not defined in the spec it should works as well, isn't it?

Reply:
No. Unfortunately, setting the MAC is required for a successful VM migration.

See this discussion for more information: https://github.com/kubevirt/kubevirt/pull/2935#issuecomment-690929185

TL;DR; we need to make sure the pod's macvlan interface mac address remains the same before / after migration. Since we lack an intelligent mechanism for that, the simplest way to achieve it, is to manually specify the mac.